subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -1328,6 +1486,45 @@ SHELL_STATIC_SUBCMD_SET_CREATE(wifi_commands,
 		"-f: Force to use this regulatory hint over any other regulatory hints\n"
 		"Note: This may cause regulatory compliance issues, use it at your own risk.",
 		cmd_wifi_reg_domain),
+	SHELL_CMD(mode, NULL, "mode operational setting\n"
+		"parameters:\n"
+		"<Interface index: interface index of the interface>\n"
+		"<Mode set value: bitmap of values.>\n"
+		"Multiple values can be set simultaneously\n"
+		"the mode support check is to be performed by lower layers\n"
+		"bit(0) = Station mode\n"
+		"bit(1) = Monitor mode\n"
+		"bit(2) = TX-Injection mode\n"
+		"bit(3) = Promiscuous mode\n"
+		"bit(4) = AP mode\n"
+		"bit(5) = STA-AP mode\n"
```

Comment:
Are all the bit combinations supported at the same time, like can we set station and AP mode at the same time, is that different from STA-AP mode?
Giving the bitmask is not very user friendly and was wondering would there be some better way to set the mode.

Reply:
The option is provided to be able to set varying bit combinations at the same time. the lower layer can decide whether that combination is a valid option or not and if the lower layer supports it. Different implementations might support different combinations. The upper layer provides a means to choose a combination. The lower layer can choose to state whether the chosen combination is valid or not

Reply:
have replaced hexadecimal based values with getopt_long based input parameters